borsh/tests/test_generic_struct.rs

```diff
@@ -133,6 +169,17 @@ where
     another: V,
 }
 
+#[allow(unused)]
+#[derive(BorshSerialize)]
+struct ParametrizedWronDerive<T, V>
+where
+    T: TraitName,
+{
+    #[borsh(bound(serialize = "<T as TraitName>::Associated: borsh::ser::BorshSerialize"))]
+    field: <T as TraitName>::Associated,
```

Comment:
`serde` does not do this magic, it just requires the developer to explicitly state the bounds:

```rust
trait TraitName {
    type Associated;
}

#[derive(serde::Serialize)]
struct ParametrizedWronDerive<T, V>
where
    T: TraitName,
    // <T as TraitName>::Associated: serde::Serialize // without this line it has a compilation error
{
    field: <T as TraitName>::Associated,
    another: V,
}
```

<img width="970" alt="image" src="https://github.com/near/borsh-rs/assets/304265/0ca58d40-a0c7-4839-a93e-36b9a037c3a9">

I think it is a sane way to implement it - let it raise compilation error and require the developer to explicitly state the bound. What do you think?

Reply:
Ok, I see that serde also has [`#[serde(bound)]` syntax](https://serde.rs/field-attrs.html#bound).

P.S. @dj8yfo mentioned to me offline that it is useful when the struct is for all types of parameters and we don't want to restrict it to be borsh-serializable.